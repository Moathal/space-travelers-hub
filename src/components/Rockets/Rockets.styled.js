import styles from 'styled-components';

export const RocketCard = styles.div`
  max-width: 100vw;
  min-height: 16rem;
  display: flex;
  padding: 0.5 rem;
  margin-bottom: 1rem;
  align-items: center;
`;

export const DetailsBlock = styles.div`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: flex-start;
`;

export const ReserveButton = styles.button`
  width: 8.5rem;
  height: 3.25rem;
  color: ${(props) => (props.reserved ? 'gray' : 'white')};
  background-color: ${(props) => (props.reserved ? 'transparent' : '#017BFE')};
  padding: 1rem;
  border: ${(props) => (props.reserved ? '1px solid gray' : 'none')};
  border-radius: 0.3rem;
  font-size: ${(props) => (props.reserved ? '0.7rem' : '0.9rem')};
`;

export const IMG = styles.img`
  width: 30%;
  height: auto;
  margin-top: 2rem;
  margin-bottom: 2rem;
  margin-right: 1rem;
  margin-left: 0.5rem;
`;

export const Badge = styles.small`
  padding: 0.25rem;
  margin-right: 3px;
  color: white;
  border-radius: 0.2rem;
  background-color: #017BFE;
  display: ${(props) => (props.reserved ? 'inline-block' : 'none')};
`;
