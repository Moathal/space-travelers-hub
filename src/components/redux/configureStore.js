import { configureStore } from '@reduxjs/toolkit';
import missionsSlice from './missions';
import rocketsReducer from './rockets';

export default configureStore({
  reducer: {
    rockets: rocketsReducer,
    missions: missionsSlice,
  },
});
