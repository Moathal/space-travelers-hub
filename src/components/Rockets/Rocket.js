import React from 'react';
import { useDispatch } from 'react-redux';
import { reserve } from '../redux/rockets';
import {
  RocketCard,
  ReserveButton,
  IMG,
  DetailsBlock,
  Badge,
} from './Rockets.styled';

function Rocket({
  img,
  name,
  description,
  id,
  reserved,
  reservation,
}) {
  const dispatch = useDispatch();
  const handleClick = () => {
    dispatch(reserve(id));
  };
  return (
    <RocketCard>
      <IMG src={img} alt="Rocket Image" />
      <DetailsBlock>
        <h2>{name}</h2>
        <p>
          <Badge reserved={reserved}>Reserved</Badge>
          {description}
        </p>
        <ReserveButton type="button" onClick={handleClick} reserved={reserved}>
          {reservation}
        </ReserveButton>
      </DetailsBlock>
    </RocketCard>
  );
}

export default Rocket;
