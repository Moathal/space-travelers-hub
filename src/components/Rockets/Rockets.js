import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Rocket from './Rocket';
import { rocketsThunk } from '../redux/rockets';

function Rockets() {
  const dispatch = useDispatch();
  const rocketsList = useSelector((state) => state.rockets);
  useEffect(() => {
    if (rocketsList.length === 0) dispatch(rocketsThunk());
  }, [dispatch]);
  return (
    <div>
      {rocketsList.map((rocket) => (
        <Rocket
          img={rocket.flickr_images[0]}
          name={rocket.rocket_name}
          description={rocket.description}
          key={rocket.id}
          id={rocket.id}
          reserved={rocket.reserved}
          reservation={rocket.reservation}
        />
      ))}
    </div>
  );
}

export default Rockets;
