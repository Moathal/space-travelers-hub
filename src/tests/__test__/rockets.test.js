import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from '../../components/redux/configureStore';
import Rockets from '../../components/Rockets/Rockets';

it('renders Rockets correctly', () => {
  const tree = renderer
    .create(
      <Provider store={configureStore}>
        <Rockets />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
