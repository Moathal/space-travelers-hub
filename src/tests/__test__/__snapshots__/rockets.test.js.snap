// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Rockets correctly 1`] = `<div />`;
